### PR TITLE
Fcn layers

### PR DESCRIPTION
**Pull request by MarvinTeichmann**
This repository provides the layers to implement [FCNs](http://arxiv.org/abs/1411.4038) into tflearn. It contains:
1.) The upscore layer.
    A "deconvolution" layer which is initialized as bilinear upsamling filter.
2.) The loss function.

The upscore layer is similar to `upsample_2d` with the following differences:
- the upsampling operation is trainable 
- it is initialized to bilinear upsampling (opposed to nearest neighbor)

 Be aware of potential Bugs, I have used those layers in my own code but did not test them es part of tflearn. For reference see also MarvinTeichmann/tensorflow-fcn#1. I implemented the changes discussed there (i.e. moved `weak_cross_entropy_2d` and renamed variables).

### CODE REVIEW

**Comment by aymericdamien** [comment]
That looks great! Thanks for the contribution :)
